src/js_lexer.zig

```diff
@@ -2012,8 +2015,9 @@ fn NewLexer_(
                             }
                         }
 
+                        // @bun must be the first pragma. it may have spaces before it.
                         if (lexer.bun_pragma == .none and strings.hasPrefixWithWordBoundary(chunk, "bun")) {
```

Comment:
This must be a bug in hasPrefixWithWordBoundary. That is supposed to only apply when there is a space or tab character between. it does another pass through the input to check.